### PR TITLE
Add user profile

### PR DESCRIPTION
**Pull request by jcompagni10**

### CODE REVIEW

**Comment by dib542** [comment]
I would like to be able to paginate the limit orders and user deposits. There is a small risk of some users with very large amounts of deposits or trades may cause this endpoint to timeout or reach some other payload size/memory limit (at the very least it may become a slow request whereas paginated pages can always be reasonably quick). So I feel like its just good practice to be able to access the individual lists. I would actually prefer to receive each paginated lists in separate calls and then collect them together in the front end just to avoid this concern.

so it would be nice to have the endpoints:
- `GET/NicholasDotSol/duality/dex/get_user_positions/{address}/pool_deposits`
- `GET/NicholasDotSol/duality/dex/get_user_positions/{address}/limit_orders`

or similar, could be keyed under a `user_profile` path or the paths could be reversed (something like `GET/NicholasDotSol/duality/dex/get_user_positions/user_profile/pool_deposits/{address}` would also work)

**Comment by dib542** [comment]
I am a little unclear on the naming of the property `totalShares` on the `PoolDeposits` object.

I've determined (in the next comment) that I only require the user's owned share amount to be returned (alongside `pairId`, `tickIndex`, and `feeIndex`).

That might be what `totalShares` is (I think that's what this is [here](https://github.com/duality-labs/duality/pull/261/files#diff-358d07cf25ba9c02cc480320a7cb111a6b1802d77fe129a816c3b2e1157e8066R67)), but I'm finding that confusing as `totalShares` sounds like the total shares existing in the tick pool from all addresses. And we did previously have a `totalShares` property representing all addresses shares on the previous version of the API ([v0.1.0](https://github.com/duality-labs/duality/blob/v0.1.0/docs/static/openapi.yml)) in the tick map response of `GET/NicholasDotSol/duality/dex/tick_map`. 

So, I would like either a clarification of `totalShares` or to have it renamed to `userShares` or `sharesOwned` or something else a little less ambiguous.

**Comment by dib542** [comment]
On the frontend I have a bit of an awkward duplication of the backend logic to compute "upper" and "lower" ticks from a user's deposited "mid" tick index and fee index in a function called [`getVirtualTickIndexes`](https://github.com/duality-labs/duality-web-app/blob/v0.1.58/src/pages/MyLiquidity/MyLiquidity.tsx#L80-L92) ("virtual" was probably not a great naming choice on my part). And while it is a simple calculation it is awkward to place this logic on the front end, as it assumes this is how the calculation happens on the backend and uses a hardcoded conversion of tickspacing as each tick space being of `tickIndex / previousTickIndex = 1 + 1/10000` (or `tickIndex = previousTickIndex * (1 + 1/10000)`)

I used this function (against backend v0.1.0) most importantly to be able to combine with the shares data existing on the tick map object (which has since [been removed before v0.1.1](https://github.com/duality-labs/duality/pull/236/files#diff-f279f1357ae0fcc8d8d331c4899b928296720ba85da48b94e38d084867dbd068L45694-L45700)) to be able to determine the theoretical amount (and value) of tokens the user "holds" with their shares: ie. the amount of token0 and token1 they would receive if they were to withdraw their liquidity at the current pair's liquidity state.

To be able to calculate the same now (the user's "held" amount of token0 and token1) without this calculation I would require these things:
- a user's amount of shares in each pool (as noted in previous comment this might be the `totalShares` value, but I would prefer if it were named more like `sharesOwned`)
- an `lowerTickIndex` and `upperTickIndex` for each user share deposit object (to replace my `getVirtualTickIndexes` function)
- an assumption of how to calculate a user's reserves0 and reserves1 from sharesOwned (to replace the lack of `totalShares` data on tick pool objects)
    - I believe a shares amount is equal to an amount of tokens held in a pool's fee tier in token0 units
        - eg. `totalShares = reserves0 + reserves1 / price0to1`
    - when a user withdraws liquidity it is done it a way to preserve the current reserves ratio:
        - eg. removing 3 shares at a reserves ratio of 2 could look like (at the tickIndex of 0)
            - token0: 4 shares -> 3 shares
            - token1: 8 shares -> 6 shares
    - therefore a user's "held" reserve0 and reserve1 with be their owned share amount distributed to each token relative to the current tick reserve amounts

With these in place it would be possible to remove the old `getVirtualTickIndexes` logic on the frontend and calculate a user's "held" reserve0 and reserve1 by:
1. taking a user's `sharesOwned` value of a tick pool
1. of that tick pool, taking `lowerTickIndex` and `upperTickIndex` to
    1. look up the `reserves0` value on `GET/NicholasDotSol/duality/dex/tick` data of `tickIndex === lowerTickIndex`
    1. look up the `reserves1` value on `GET/NicholasDotSol/duality/dex/tick` data of `tickIndex === upperTickIndex`
    1. look up the `price0To1` value on `GET/NicholasDotSol/duality/dex/tick` data of `tickIndex === upperTickIndex`
1. calculating each of the user's reserves with
    1. `percentReserves0 = reserves0/(reserves0 + reserves1 / price0to1)`
    1. `percentReserves1 = (reserves1 / price0to1)/(reserves0 + reserves1 / price0to1)`
    1. `userReserves0` = `sharesOwned * percentReserves0`
    1. `userReserves1` = `sharesOwned * percentReserves1 / price0to1`

---

Side note:

At this point it might seem like adding `userReserves0` and `userReserves1` amounts onto the `GET/NicholasDotSol/duality/dex/get_user_positions/{address}/pool_deposits` object would be a possible alternative solution, and technically it is, but it's very impractical because I often need to get and keep the general tick data up to date anyway. For example on the MyLiquidity page I show both the user's ticks and the general tick pool's ticks in the background for comparisons sake, so a premade backend calculation of `userReserves0` and `userReserves1` would not particularly save time and it would be harder to keep this in sync with the general market data instead of deriving from it.

It is also just better for the `GET/NicholasDotSol/duality/dex/get_user_positions/{address}/pool_deposits` endpoint to remain concerned only with the user's pool deposit (and the response to not change with changing market conditions, only changing with edits made by the user) so that I can call for the user's deposits information for that information only. The current market information (including whether a user's shares resolve to token0 or token1) should be kept up to date by continually watching just the ticks data.

---

To summarize

I require:
- the user's share amount to be returned on the user's deposit object

I do not require:
- the total share amount of a tick pool

I would like:
- a confirmation of that `totalShares = reserves0 + reserves1 / price0to1` (all reserves in token0 units)
- `lowerTickIndex` and `upperTickIndex` returned on user deposit objects (to replace my FE `getVirtualTickIndexes`)

File: x/dex/keeper/limit_order_tranche_user.go [hunk 1]
```diff
@@ -80,3 +80,30 @@ func (k Keeper) GetAllLimitOrderTrancheUser(ctx sdk.Context) (list []types.Limit
 
 	return
 }
+
+func LimitOrderTrancheUserAddressPrefix(address string) []byte {
+
+	key := types.KeyPrefix(types.LimitOrderTrancheUserKeyPrefix)
+	addressBytes := []byte(address)
```
Comment (teddyknox) [inline x/dex/keeper/limit_order_tranche_user.go:87]: can this size be statically defined?

**Review by teddyknox — COMMENTED** [review]
Does this return all positions or all open positions?

File: proto/duality/dex/query.proto [hunk 2]
```diff
@@ -92,6 +94,21 @@ service Query {
 		option (google.api.http).get = "/NicholasDotSol/duality/dex/limit_order_tranche";
 	}
 
+// Queries a list of GetUserPositions items.
+	rpc GetUserPositions(QueryGetUserPositionsRequest) returns (QueryGetUserPositionsResponse) {
```
Comment (teddyknox) [inline proto/duality/dex/query.proto:98]: Should have `All` in the name

**Comment by teddyknox** [comment]
@jcompagni10 what about pagination?

**Review by teddyknox — APPROVED** [review]
